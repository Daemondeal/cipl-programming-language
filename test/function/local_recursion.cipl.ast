pass
(block
    (proc fib (params n) (body
        (if (< n 2) (then (block
            (return n)
        ))
        (return (+ (call fib (- n 1)) (call fib (- n 2))))
    ))
    (expression (call println (call fib 8)))
)

