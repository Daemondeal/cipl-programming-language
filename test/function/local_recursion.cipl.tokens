Token          Lexeme         Line 
PASS           pass           1    
LINE_END       \n             2    
INDENT                        2    
PROC           proc           2    
IDENTIFIER     fib            2    
LEFT_PAREN     (              2    
IDENTIFIER     n              2    
RIGHT_PAREN    )              2    
COLON          :              2    
LINE_END       \n             3    
INDENT                        3    
IF             if             3    
IDENTIFIER     n              3    
LESS           <              3    
NUMBER         2              3     2
COLON          :              3    
LINE_END       \n             4    
INDENT                        4    
RETURN         return         4    
IDENTIFIER     n              4    
LINE_END       \n             5    
DEDENT                        6    
RETURN         return         6    
IDENTIFIER     fib            6    
LEFT_PAREN     (              6    
IDENTIFIER     n              6    
MINUS          -              6    
NUMBER         1              6     1
RIGHT_PAREN    )              6    
PLUS           +              6    
IDENTIFIER     fib            6    
LEFT_PAREN     (              6    
IDENTIFIER     n              6    
MINUS          -              6    
NUMBER         2              6     2
RIGHT_PAREN    )              6    
LINE_END       \n             7    
DEDENT                        8    
IDENTIFIER     println        8    
LEFT_PAREN     (              8    
IDENTIFIER     fib            8    
LEFT_PAREN     (              8    
NUMBER         8              8     8
RIGHT_PAREN    )              8    
RIGHT_PAREN    )              8    
LINE_END       \n             8    
DEDENT                        8    
EOF                           8    
