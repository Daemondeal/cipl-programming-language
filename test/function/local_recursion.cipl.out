::stdout::
21

::stderr::
